### PR TITLE
feat(KFLUXDP-99): Add Tekton task that instrument Golang source code with sealights agents

### PR DESCRIPTION
**Pull request by flacatus**
# Description

This pull request adds a new Tekton task that makes it easier to instrument Go code with Sealights for quality analytics. The task automates retrieving the source code, running the Sealights instrumentation, and creating a new trusted artifact with the instrumented code.

A README file is included with clear instructions and an example pipeline to help you get started quickly

### CODE REVIEW

File: tasks/sealights/sealights-go-instrumentation/0.1/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Tekton Task: Sealights Go Instrumentation
+
+Please read/update [test metadata docs](../../../../docs/qe-available-tasks/sealights/go-instrumentation/go-instrumentation.md) from docs folder.
```
Comment (psturc) [inline tasks/sealights/sealights-go-instrumentation/0.1/README.md:3]: would it make sense to start documenting tasks/stepactions in build-definitions style? i.e. README.md in every version, which is useful when something changes in the task and the documentation needs to change.

example: https://github.com/konflux-ci/build-definitions/tree/main/task/sast-snyk-check/0.3

Comment (flacatus) [reply tasks/sealights/sealights-go-instrumentation/0.1/README.md:3]: I think yes. Also I think if we use gitbook the .md files should be under docs foldee

File: docs/qe-available-tasks/sealights/go-instrumentation/go-instrumentation.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+3. **Creates** a new trusted artifact containing the instrumented code.
+
+The task can be triggered by different events (e.g., Pull Request, Push) and allows users to exclude specific Go packages from the instrumentation process.
+
+## Parameters
+
+| Name                  | Type     | Default       | Description                                                                                   |
+|-----------------------|----------|---------------|-----------------------------------------------------------------------------------------------|
+| `source-artifact`     | `string` | -             | The Trusted Artifact URI pointing to the source code.                                         |
+| `go-version`          | `string` | -             | The Go version to use (e.g., `1.21.3`).                                                       |
```
Comment (psturc) [inline docs/qe-available-tasks/sealights/go-instrumentation/go-instrumentation.md:20]: would be maybe worth to document that the supported versions of Go are based on what's available in the [catalog](https://catalog.redhat.com/software/containers/ubi8/go-toolset/5ce8713aac3db925c03774d1)?

File: tasks/sealights/sealights-go-instrumentation/0.1/sealights-go-instrumentation.yaml [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    - name: source-artifact
+      description: The Trusted Artifact URI pointing to the artifact with
+        the application source code.
+      type: string
+    - name: go-version
+      type: string
+      description: "The Go version to use with the 'ubi8/go-toolset' image, in the format '1.x.y' (e.g., '1.21.3')."
+    - name: sealights-secret
+      type: string
+      description: "The name of the Openshift secret containing Sealights credentials."
```
Comment (psturc) [inline tasks/sealights/sealights-go-instrumentation/0.1/sealights-go-instrumentation.yaml:34]: maybe mention that the key `token` is expected and point to the docs which describes how to get it?

File: docs/qe-available-tasks/sealights/go-instrumentation/go-instrumentation.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    - name: instrument-go-code
+      taskRef:
+        resolver: git
+        params:
+          - name: url
+            value: https://github.com/konflux-ci/tekton-integration-catalog.git
+          - name: revision
+            value: main
+          - name: pathInRepo
+            value: tasks/sealights/sealights-go/0.1/sealights-go.yaml
```
Comment (psturc) [inline docs/qe-available-tasks/sealights/go-instrumentation/go-instrumentation.md:110]:
```suggestion
            value: tasks/sealights/sealights-go-instrumentation/0.1/sealights-go-instrumentation.yaml
```

File: docs/qe-available-tasks/sealights/go-instrumentation/go-instrumentation.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        - name: sealights-secret
+          value: "$(params.sealights-secret)"
+        - name: component
+          value: "$(params.component)"
+        - name: oci-storage
+          value: "$(params.oci-storage)"
+        - name: exclude-packages
+          value:
+            - "github.com/modern-go/concurrent"
+            - "github.com/go-logr/logr"
```
Comment (psturc) [inline docs/qe-available-tasks/sealights/go-instrumentation/go-instrumentation.md:125]: should `revision` param be added here?

**Review by psturc — COMMENTED** [review]
couple of minor things, overall lgtm 👍

File: docs/qe-available-tasks/sealights/go-instrumentation/go-instrumentation.md (hunk 2 above)
Comment (flacatus) [reply docs/qe-available-tasks/sealights/go-instrumentation/go-instrumentation.md:20]: good point!

File: tasks/sealights/sealights-go-instrumentation/0.1/sealights-go-instrumentation.yaml (hunk 3 above)
Comment (flacatus) [reply tasks/sealights/sealights-go-instrumentation/0.1/sealights-go-instrumentation.yaml:34]: cool. Let me find the docs.

**Review by psturc — APPROVED** [review]

**Pull request merged**